components/request_otr/browser/request_otr_service.cc

```diff
@@ -53,6 +53,49 @@ bool RequestOTRService::ShouldBlock(const GURL& url) const {
   return false;
 }
 
+bool RequestOTRService::OfferedOTR(const GURL& url) {
+  if (!url.SchemeIsHTTPOrHTTPS()) {
+    return false;
+  }
+  return base::Contains(offered_otr_cache_,
+                        net::URLToEphemeralStorageDomain(url));
+}
+
+bool RequestOTRService::RequestedOTR(const GURL& url) {
+  if (!url.SchemeIsHTTPOrHTTPS()) {
+    return false;
+  }
+  return base::Contains(requested_otr_cache_,
+                        net::URLToEphemeralStorageDomain(url));
+}
+
+void RequestOTRService::SetOfferedOTR(const GURL& url) {
+  if (!url.SchemeIsHTTPOrHTTPS()) {
+    return;
+  }
+
+  offered_otr_cache_[net::URLToEphemeralStorageDomain(url)] =
+      url::Origin::Create(url);
+}
+
+void RequestOTRService::SetRequestedOTR(const GURL& url) {
+  if (!url.SchemeIsHTTPOrHTTPS()) {
+    return;
+  }
+
+  requested_otr_cache_[net::URLToEphemeralStorageDomain(url)] =
+      url::Origin::Create(url);
+}
+
+void RequestOTRService::WithdrawOTR(const GURL& url) {
+  if (!url.SchemeIsHTTPOrHTTPS()) {
+    return;
+  }
+  std::string domain = net::URLToEphemeralStorageDomain(url);
+  // offered_otr_cache_.erase(domain);
```

Comment:
?

Reply:
re-enabled in https://github.com/brave/brave-core/commit/c83ae23838b8132c40d9ce007c338abf52804ae9